src/adapter.js

```diff
@@ -12,7 +12,51 @@ export const mediaTypes = [
 ];
 
 export function detect(source) {
-  return !!source.match(/"?swagger"?:\s*["']2\.0["']/g);
+  return !!(_.isString(source)
+    ? source.match(/"?swagger"?:\s*["']2\.0["']/g)
+    : source.swagger === '2.0');
+}
+
+// Test whether a key is a special Swagger extension.
+function isExtension(value, key) {
+  return key.indexOf('x-') === 0;
+}
+
+// Test whether tags can be treated as resource groups, and if so it sets a
+// group name for each resource (used later to create groups).
```

Comment:
Swagger tags are overlapping transition groups and I thing we have to come up with support for this in the refract API description namespace.